File: dotnet/src/model/TransferInput.cs
using System.Text.Json.Serialization;

namespace MoneyTransfer;

public class TransferInput
{
    [JsonPropertyName("amount")]
    public int Amount { get; set; }
    [JsonPropertyName("fromAccount")]
    public string FromAccount { get; set; }
    [JsonPropertyName("toAccount")]
    public string ToAccount { get; set; }

    public TransferInput()
    {
        Amount = 0;
        FromAccount = "";
        ToAccount = "";
    }

    public TransferInput(int amount, string fromAccount, string toAccount)
    {
        Amount = amount;
        FromAccount = fromAccount;
        ToAccount = toAccount;
    }
}

File: ruby/models/transfer_input.rb
# frozen_string_literal: true

require 'active_support/inflector'
require 'active_support/core_ext/hash/keys'

module Models
  TransferInput = Struct.new(:amount, :from_account, :to_account) do
    def self.from_h(hash)
      symbolized = hash.deep_transform_keys { |key| key.to_s.underscore.to_sym }
      new(**symbolized.slice(:amount, :from_account, :to_account))
    end
  end
end
